instrumentation/net/http/otelhttp/internal/semconv/env_test.go

```diff
@@ -132,3 +169,73 @@ func NewTestHTTPClient() HTTPClient {
 		latencyMeasure:       &testInst{},
 	}
 }
+
+func BenchmarkRecordMetricsPooled(b *testing.B) {
```

Comment:
These two benchmarks won't provide any meaningful value.
You need to split the test cases into sub benchmarks, so each of them provides its own value.

`ReportAllocs` and `ResetTimer` need to be called before the action we want to call, since they specify the data we want to record and when we want to start recording.

You may want to look at https://dave.cheney.net/2013/06/30/how-to-write-benchmarks-in-go as a guidance on how to write benchmarks.
This repository and the opentelemetry-go one also have several benchmarks you can look at as examples.

Reply:
I have fixed the benchmarks and am now getting results:

```
goos: darwin
goarch: arm64
pkg: go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp/internal/semconv
cpu: Apple M1 Max
BenchmarkHTTPServerRequest
BenchmarkHTTPServerRequest-10      	 3890485	       287.3 ns/op	     712 B/op	       2 allocs/op
BenchmarkRecordMetricsPooled
BenchmarkRecordMetricsPooled-10    	  552594	      2192 ns/op	    4073 B/op	      28 allocs/op
BenchmarkRecordMetrics
BenchmarkRecordMetrics-10          	  537703	      2152 ns/op	    4104 B/op	      30 allocs/op
PASS
ok  	go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp/internal/semconv	5.713s
```

The number of allocs/op grew linearly for both benchmarks with double the amount of calls resulting in 56 Pooled and 60 NotPooled allocs/op.

The Pooled implementation is slightly faster and has slightly fewer bytes and allocs per operation. Do you want me to keep this benchmark in this file or another file or just move forward with replacing the `RecordMetrics()` with the pooling code?

Reply:
You only changed one benchmark. They need to all be changed.
Also, rather than setting up a brand new method (which kind of borks the test. We need to be able to benchmark the actual change that will be made), could you setup one benchmark, and run it against both the main branch, and this PR?

You can also use the [benchstat](https://pkg.go.dev/golang.org/x/perf/cmd/benchstat) tool to make the comparison between two runs of the same benchmark more readable.

Basically, this PR should include:

* The change you wish to make, as it will be in the end.
* One benchmark, that tests that change's performance.
* In the PR description, the benchstat comparison of that benchmark against the main branch, and your branch.

Reply:
Got it. I intended to remove the extra methods once there was proof that pooling was an improvement. I will update the PR to fit this structure and move it out of draft.